crates/wasmparser/src/readers/core/types.rs

```diff
@@ -333,15 +333,6 @@ impl RefType {
     const fn from_u32(x: u32) -> Self {
         debug_assert!(x & (0b11111111 << 24) == 0);
 
-        // if indexed, kind must be struct/array/func
-        debug_assert!(
-            x & Self::INDEXED_BIT == 0
-                || matches!(
-                    x & Self::KIND_MASK,
-                    Self::FUNC_KIND | Self::ARRAY_KIND | Self::STRUCT_KIND
-                )
-        );
```

Comment:
If the RefType refers to an indexed type, the 2-bit area masked by `Self::KIND_MASK` can be anything. `00` means "we don't know the kind".

Reply:
I think we might actually want to completely remove the kind from indexed `RefType`s, since it is redundant with the information in the actual type section. Maintaining the same information in multiple places just allows the possibility for things to get out of sync. Can you make a follow up to remove the indexed kind and actually index into the type section instead everywhere where we are currently using the kind?